clock_exercise/Clock.qml

```diff
@@ -0,0 +1,241 @@
+                source: "images/ic-clock-hand_hours.svg"
+
+                transform: Rotation{
+                    id: minuteHandRotation
+
+                    angle: clockManager !== null? clockManager.minutesClockHandAngle: 0
+                    origin.x: minuteHand.height/2
+                    origin.y: minuteHand.height/2
+                }
+            }
+
+            Image {
+                id: clockHandsCenter
+
+                width: 22
+                height: width
+                source: "images/ic-clock-hands_center.svg"
+                anchors.centerIn: parent
+            }
+        }
+    }
+
+    Rectangle {
+        id: timerButton
+
+        x: 18
+        y: 690
+        width: 190
+        height: 64
+        color: timerButtonMouseArea.containsMouse? "#FCB647": "#151B2E"
+        border.color : "#FCB647"
+        border.width : 1
+        radius: 32
+
+        Behavior on color {
+            ColorAnimation {
+                duration: timeOfDissolve
+            }
+        }
+
+        Text {
+            id: timerButtonText
+
+            width: 96
+            height: 36
+            text: "TIMER"
+            color: timerButtonMouseArea.containsMouse? "#151B2E": "#FCB647"
+            anchors.centerIn: parent
+            font.pixelSize: 30
+            horizontalAlignment: Text.AlignHCenter
+            font.family: buenosAiresSemiBold.name
+            font.styleName: "SemiBold"
+
+            Behavior on color {
+                ColorAnimation {
+                    duration: timeOfDissolve
+                }
+            }
+
+        }
+
+        MouseArea {
+            id: timerButtonMouseArea
+
+            hoverEnabled: true
+            anchors.fill: parent
+            onClicked: clockPage.timerButtonClicked()
+        }
+    }
+
+    Rectangle {
+        id: alarmButton
+
+        x: 272
+        y: 690
+        width: 190
+        height: 64
+        color: alarmButtonMouseArea.containsMouse? "#FCB647": "#151B2E"
+        border.color : "#FCB647"
+        border.width : 1
+        radius: 32
+
+        Behavior on color {
+            ColorAnimation {
+                duration: timeOfDissolve
+            }
+        }
+
+        Text {
+            id: alarmButtonText
+
+            width: 96
+            height: 36
+            text: "ALARM"
+            color: alarmButtonMouseArea.containsMouse? "#151B2E": "#FCB647"
+            anchors.centerIn: parent
+            font.pixelSize: 30
+            horizontalAlignment: Text.AlignHCenter
+            font.family: buenosAiresSemiBold.name
+            font.styleName: "SemiBold"
+
+            Behavior on color {
+                ColorAnimation {
+                    duration: timeOfDissolve
+                }
+            }
+        }
+
+        MouseArea {
+            id: alarmButtonMouseArea
+
+            hoverEnabled: true
+            anchors.fill: parent
+            onClicked: clockPage.alarmButtonClicked()
+        }
+    }
+}
```

Comment:
Diciamo che quando vedo 240 righe di QML per la componente Clock mi insospettisco :P
Una delle mie `rule of thumb` è che se in QML supero le 100 righe, devo chiedermi se davvero è necessario (magari la componente o la pagina effettivamente è complessa) oppure se ho sbagliato qualcosa nella modularizzazione della componente stessa.